Source/Core/Common/CommonPaths.cpp

```diff
@@ -0,0 +1,536 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Common/CommonPaths.h"
+#include "Common/FileUtil.h"
+#include "Common/Logging/Log.h"
+
+#ifdef _WIN32
+#include <windows.h>
+#include "Common/StringUtil.h"
+#else
+#include <limits.h>
+#include <unistd.h>
+#endif
+
+#ifdef __APPLE__
+#include <CoreFoundation/CFBundle.h>
+#include <CoreFoundation/CFString.h>
+#include <CoreFoundation/CFURL.h>
+#include <sys/param.h>
+#endif
+
+// Shared data dirs (Sys and shared User for Linux)
+constexpr auto SYSDATA_DIR =
```

Comment:
I wrote some example code:

``` c++
#include <iostream>
#include <string>
#include <type_traits>

int main() {
  constexpr auto what_am_i = "What am I";
  std::cout<<"Type is: " << typeid(what_am_i).name() << std::endl;

  return 0;
}
```
Compiled with `clang++ -std=c++14 auto_string_test.cpp -o auto_string_test.o` on a Macbook 2014 Early 14 Air, and run with `./auto_string_test.o | c++filt`, yields `char const*`

However, `constexpr std::strings` are not possible due to them not being literals, so `std::string` would not work. There are two solutions; either have a compile-time c-style string which can be constructed into a `std::string` at runtime, or use `std::experimental::string_view`, which is a c++17 feature. 

So, leaving it `const char*` is the best idea imo.